src/happy_vllm/routers/schemas/functional.py

```diff
@@ -136,4 +140,30 @@ class HappyvllmCompletionResponse(CompletionResponse):
 
 
 class HappyvllmChatCompletionResponse(ChatCompletionResponse):
-    model_config = {"json_schema_extra": {"examples": [response_examples["chat_completion_response"]]}}
\ No newline at end of file
+    model_config = {"json_schema_extra": {"examples": [response_examples["chat_completion_response"]]}}
+
+
+async def update_chat_completion_request(request: Request, data: ChatCompletionRequest):
+    """
+    Updates a ChatCompletionRequest object with additional tools and settings if available.
+
+    Args:
+        request (Request): The incoming request object.
+        data (ChatCompletionRequest): The original ChatCompletionRequest object to be updated.
+
+    Returns:
+        ChatCompletionRequest: The updated ChatCompletionRequest object if tools are present,
+                               otherwise returns the original ChatCompletionRequest object.
+    """
+    tools : Union[dict, None] = get_tools_prompt()
+    if tools:
+        data_dict = data.dict()
+        if data_dict['tools']:
+            data_dict['tools'].extend(tools["tools"])
+        else:
+            data_dict['tools'] = tools["tools"]
+        if data_dict['top_logprobs'] == 0:
```

Comment:
[ChatCompletionRequest](https://github.com/vllm-project/vllm/blob/main/vllm/entrypoints/openai/protocol.py#L96) have top_logprobs(int) and logprobs(bool) optional and a method(pydantic) to check if top_logprobs then logprobs should be True.
When I get the ChatCompletionRequest.data instantiate with top_logprobs=None and logprobs=None, we top_logprobs=0 and logprobs=False and we have to rewritte top_logprobs with None to avoid method check error at the new ChatCompletionRequest's instaciation